android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js

```diff
@@ -17,11 +17,11 @@ browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
        });
        break;
      case 'addSerializedDoc':
-        storage.session.set(sender.contextId.toString(), message.doc);
+        browser.storage.session.set(sender.contextId.toString(), message.doc);
```

Comment:
The arguments for `browser.storage.session.set` are incorrect and I think we should be waiting (?) so suggesting an alternative. 

Also, in debugging I found that `addSerializedDoc` is called twice and `getSerializedDoc` is called twice when opening a readerview. After each closing of readerview when the user goes to open the same page in readerview the `contextId` has changed and it again calls `addSerializedDoc` and `getSerializedDoc` twice.

Unless I'm understanding incorrectly, shouldn't the correct behavior be the first opening of the readerview we add the serialized doc with a persistent contextId (or another type of UID) and then subsequent openings of this page call we just call `getSerializedDoc` and return the cached doc?
```suggestion
        let sessionContextKey = sender.contextId.toString();
        await browser.storage.session.set({sessionContextKey: message.doc});
```

Reply:
> The arguments for `browser.storage.session.set` are incorrect and I think we should be waiting (?) so suggesting an alternative.

right...

> the `contextId` has changed

looks like a pre-existing issue, but yeah...